server/__init__.py

```diff
@@ -17,32 +17,37 @@
 #  limitations under the License.
 ###############################################################################
 
-from girder import events
-from girder.api import access
-from girder.api.rest import boundHandler
+from bson import json_util
+import json
+
 from girder.constants import SettingKey
+from girder.models.folder import Folder
+from girder.models.item import Item
 from girder.models.setting import Setting
 
 from . import rest
 
 
-@access.public
-@boundHandler
-def _virtualChildItems(self, event):
-    params = event.info['params']
+originalChildItems = None
+
 
-    if params['type'] != 'folder':
-        return  # This can't be a virtual folder
-    try:
-        import girder.plugins.virtual_folders
-    except ImportError:
-        return  # If virtual folders aren't enabled, do nothing
-    params['folderId'] = event.info['id']
-    return girder.plugins.virtual_folders._virtualChildItems(event)
+def childItems(self, folder, limit=0, offset=0, sort=None, filters=None,
+               includeVirtual=False, **kwargs):
+    if not includeVirtual or not folder.get('isVirtual') or 'virtualItemsQuery' not in folder:
+        return originalChildItems(self, folder, limit=limit, offset=offset,
+                                  sort=sort, filters=filters, **kwargs)
+    q = json_util.loads(folder['virtualItemsQuery'])
```

Comment:
as far as I know the plugin handles cases in which is is an invalid JSON, like wrapping with a proper try/catch

Reply:
The stored query is validated on creation (https://github.com/girder/girder/blob/c565081e65d589e896fca4f98c8b32e9794df051/plugins/virtual_folders/girder_virtual_folders/__init__.py#L40).  In the virtual folder plugin, this is assumed and not guarded in use.